test/performance/pacers.go

```diff
@@ -24,38 +24,55 @@ import (
 	vegeta "github.com/tsenart/vegeta/lib"
 )
 
-// SteadyUpPacer is a Pacer that describes attack request rates that increases in the beginning then becomes steady.
+// steadyUpPacer is a Pacer that describes attack request rates that increases in the beginning then becomes steady.
 //  Max  |     ,----------------
 //       |    /
 //       |   /
 //       |  /
 //       | /
 //  Min -+------------------------------> t
 //       |<-Up->|
-type SteadyUpPacer struct {
+type steadyUpPacer struct {
 	// UpDuration is the duration that attack request rates increase from Min to Max.
 	UpDuration time.Duration
-	// Min is the attack request rates from the beginning.
+	// Min is the attack request rates from the beginning, must be larger than 0.
 	Min vegeta.Rate
 	// Max is the maximum and final steady attack request rates.
 	Max vegeta.Rate
+
+	slope        float64
+	minHitsPerNs float64
+	maxHitsPerNs float64
+}
+
+// NewSteadyUpPacer returns a new SteadyUpPacer with the given config.
+func NewSteadyUpPacer(min vegeta.Rate, max vegeta.Rate, upDuration time.Duration) vegeta.Pacer {
+	pacer := &steadyUpPacer{
+		Min:        min,
+		Max:        max,
+		UpDuration: upDuration,
+	}
+	pacer.slope = (hitsPerNs(max) - hitsPerNs(min)) / float64(upDuration)
```

Comment:
Inline those as well above and then you can just
`return &steadUpPacer{...}`

Reply:
Done.